src/components/videoCard/VideoCard.jsx

```diff
@@ -1,24 +1,90 @@
 import "./VideoCard.css";
-import {HiDotsVertical} from "react-icons/hi";
-import {MdPlayCircleFilled} from "react-icons/md";
-export const VideoCard=({videos})=>{
-    return(
+import { HiDotsVertical } from "react-icons/hi";
+import { MdPlayCircleFilled, MdOutlineDelete } from "react-icons/md";
+import { RiPlayListAddFill } from "react-icons/ri";
+import { AiOutlineClockCircle } from "react-icons/ai";
+import { useState, useRef, useEffect } from "react";
+import { isInList } from "./helper";
+import { useAuth } from "../../context/auth-context";
+import { useStateContext } from "../../context/state-context";
+import { removeFromWatchLater, addToWatchLater } from "../../services/index";
+import { useNavigate, useLocation } from "react-router-dom";
+import { toast } from "react-toastify";
+
+export const VideoCard = ({ videos }) => {
+    const {
+        _id: id,
+        title,
+        categoryName,
+        creator,
+        duration,
+        views,
+        thumbnail,
+    } = videos;
+
+    const [isModalActive, setModalActive] = useState(false);
+    const ref = useRef();
+    const { authState: { token } } = useAuth();
+    const { state: { watchLater }, dispatch } = useStateContext();
+
+    const isInWatchLater = isInList(watchLater, id);
+    const navigate = useNavigate();
+    const location = useLocation();
+
+
+    const watchLaterHandler = () => {
+        setModalActive(prev => !prev)
+        token
+            ? isInWatchLater
```

Comment:
instead of nested ternary operator use some object mapper or any other clean approach (Just a suggestion)